### PR TITLE
Add support for Zephyr intergartion

### PR DESCRIPTION
**Pull request by vanwinkeljan**
### Description of the feature

Add support to build LVGL as a Zephyr RTOS library and use the LVGL repository as a Zephyr module.

This changes will allow easier integration with Zephyr without the need to maintain additional changes in the downstream Zephyr specific LVGL repository. Actually once this changes are merged the downstream LVGL repository can function as a one-to-one mirror*.

Further this PR removes `src/lv_conf_zephyr.h` as this file should be maintained in the Zephyr repo it self

\* Note that the Zephyr policy mandates that a mirror is  maintain for each repo that is included as a module into Zephyr

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Update CHANGELOG.md
- [ ] Update the documentation

### CODE REVIEW

**Review by embeddedt — APPROVED** [review]

**Comment by embeddedt** [comment]
We're unlikely to hear from @kisvegabor for the next week or so as he is on Christmas vacation.

Can you please add a line about this to CHANGELOG.md as well?

**Comment by vanwinkeljan** [comment]
@embeddedt 

> Can you please add a line about this to CHANGELOG.md as well?

Done

**Comment by vanwinkeljan** [comment]
@gmarull could you have a look at this PR from the Zephyr side?

**Review by kisvegabor — APPROVED** [review]
Amazing! Looks good to me!

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,6 +32,34 @@ if (CONFIG_LV_ATTRIBUTE_FAST_MEM_USE_IRAM)
     target_compile_definitions(${COMPONENT_LIB} PUBLIC "-DLV_ATTRIBUTE_FAST_MEM=IRAM_ATTR")
 endif()
 
+elseif(ZEPHYR_BASE)
+
+if(CONFIG_LVGL)
+
+set(ZEPHYR_CURRENT_LIBRARY lvgl)
+include_directories(${ZEPHYR_BASE}/lib/gui/lvgl)
```
Comment (gmarull) [inline CMakeLists.txt]:
```suggestion
zephyr_include_directories(${ZEPHYR_BASE}/lib/gui/lvgl)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -32,6 +32,34 @@ if (CONFIG_LV_ATTRIBUTE_FAST_MEM_USE_IRAM)
...
+
+target_include_directories(lvgl INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (gmarull) [inline CMakeLists.txt:41]: is `lvgl` target defined somewhere else?

**Review by gmarull — COMMENTED** [review]
Looks good to me, left a couple of questions

File: CMakeLists.txt (hunk 1 above)
Comment (vanwinkeljan) [reply CMakeLists.txt]: Actually the two lines are no longer needed, I will remove them

File: CMakeLists.txt (hunk 2 above)
Comment (vanwinkeljan) [reply CMakeLists.txt:41]: Yes the `lvgl` target is defined on the zephyr side  (`lib/gui/lvgl/CMakeLists.txt`)

**Review by gmarull — APPROVED** [review]

**Comment by kisvegabor** [comment]
Merged, thank you!

**Pull request merged**